### DISCUSSION TITLE
Idea for the "Make enhancements to user profiles" project

### DISCUSSION THREAD

**Discussion by khabdrick** [unstructuredstudio/zubhub#411] [Ideas]
For the profile enhancement project, my idea is, directly below Projects, Followers etc. We can have a graph of the number of projects on a weekly basis and the number of views on a weekly basis, separately. You can see a demonstration in the images below.

I think we can get the start and end of week data using `datetime` module then pass it to the frontend. There are functions that handle the number of views and number of projects so that part should be easy.

The library that will come in that we can use on the front end with React is maybe Plotly.js or CanvasJS.
For the badge for “project of the week” I suggest can be assigned to a project with the highest number of views+claps.

Please add more suggestions in the comment section!

![2022-04-18_09-16](https://user-images.githubusercontent.com/57852546/163787972-321a93f5-9fb6-4971-94be-83ac548183d3.png)

![2022-04-18_10-03](https://user-images.githubusercontent.com/57852546/163788007-a755b204-9416-41a6-bbe3-48a92277f155.png)

**Comment by tuxology** [comment]
Thanks for this @khabdrick We need to think a bit more in depth there. Here are a few suggestions and queries:

- While number of views per week looks like a good metric, number of projects per week does not really makes sense. For a vast number of people, this metric will be 0. For a few it may be 1 or 2 since that's as many projects you can realistically create in a week. Therefore, this metric should not be plotted (as the graphs will look empty most of the time) but instead we can have a simple counter which increments. With each increment (lets say 10) we change some style, add a badge etc. adjacent to it when he badge feature is introduced later on
- For a start, storing a week's worth of metrics stored per user looks fine to me. Eventually we will store more longer _time series_ data so whatever design we come up with needs to be flexible to accommodate that. As I said before, lets come up with 3-4 concrete metrics that are actually useful and interesting for users and then think about how to represent them. The goal of these metrics is to continuously inspire the users and to allow them to see their status in the Zubhub ecosystem. How would these users (creators for now) benifit from each new thing we add on their profile?
-  Lets sketch some early designs of how these metrics you come up with would look on the current profile. Where would you place them? Rough sketches drawn over current profile screenshots can be ok to begin with. 
- The badge part will come later I think and can be a separate discussion

**Reply by khabdrick** [reply]
@tuxology how many metrics will be on the profile itself?

**Reply by tuxology** [reply]
As of now, I can think of a few at least:

1. Follower Count
2. Project Count
3. Total project views/week
4. Total activity instances/week (an activity instance could be comments left + projects liked)

**Reply by souad988** [reply]
I think also the way all buttons links and sections are displayed can be improved, by adding a side navBar with a list of icons and explanatory text for each of them

**Reply by khabdrick** [reply]
> As of now, I can think of a few at least:
> 
> 1. Follower Count
> 2. Project Count
> 3. Total project views/week
> 4. Total activity instances/week (an activity instance could be comments left + projects liked)
![2022-04-19_15-06](https://user-images.githubusercontent.com/57852546/164022879-289ea542-cb5b-46e0-87bc-014609c29493.png)
Sorry, I'm not really familiar with designing tools. I think the metrics  can be above the about me section.
There's always a limited number of comments, that's why I'll suggest like+views instead.
@tuxology

**Reply by khabdrick** [reply]
> I think also the way all buttons links and sections are displayed can be improved, by adding a side navBar with a list of icons and explanatory text for each of them.

@souad988 I might be wrong, but I don't think this is within the scope of the project.

**Reply by souad988** [reply]
I think this is part of the internship work, so it is a big task not a small issue so we may suggest more improvements  if i'm not wrong what do you think @khabdrick

**Reply by Deepanshu039** [reply]
> As of now, I can think of a few at least:
> 
> 1. Follower Count
> 2. Project Count
> 3. Total project views/week
> 4. Total activity instances/week (an activity instance could be comments left + projects liked)

Also an option to add custom image🧑 to the profile picture should be there along with the avatar.
--possible reasons to do it :-
-->> If a creator want to add their own photo.
-->> If a creator want to make any project image as their default profile image.
-->> If a creator want to display any accomplishment..

**Comment by souad988** [comment]
I think putting all profile bio and info in the sidebar with the edit and delete buttons, and all the metrics in the middle with previews of best viewed projects and comments can be good for a better user experience , something near to github profile, and we can use the circular progress indicator component from react for the progress circle bar.
![github profile](https://user-images.githubusercontent.com/59707859/163982718-8249f9e0-a862-4410-bc56-b7a6e2da7b54.PNG)

**Comment by Deepanshu039** [comment]
An option for connecting details of creators can be added on the user profile page such as a Twitter handle or Gmail id etc.
>> beneficial in cases when someone wants to connect with the creator and share some ideas for a new project.
>> someone likes to collaborate on a new project with the creator.
>> or someone who suggests some improvements in a project can privately share with creators(irrespective of the comment section).

**Reply by khabdrick** [reply]
@Deepanshu039 sounds good

**Reply by Deepanshu039** [reply]
> @Deepanshu039 sounds good

any suggestions would you like to add @khabdrick .

**Reply by khabdrick** [reply]
> > @Deepanshu039 sounds good
> 
> any suggestions would you like to add @khabdrick .

None at this time

**Comment by Deepanshu039** [comment]
![Screenshot (96)](https://user-images.githubusercontent.com/92849065/166471332-df176998-f990-4a3b-8f88-fe90bcbec392.png)

I am not sure but this follow button on the right side just doesn't feel right maybe it's the alignment of the button or maybe the button is too isolated on the page. My suggestion is if we could just place it somewhere else on the page or if could just make it look more consistent in the current position. Also, its alignment is not right with the about me component.
